helpers/setURLQueryParam.js

```diff
@@ -17,7 +17,7 @@ const factory = () => {
             throw new TypeError("Invalid query parameter value passed to setURLQueryParam");
         }
 
-        const url = new URL(string);
+        const url = NodeJsURL ? new NodeJsURL(string) : new URL(string);
```

Comment:
I'd move out this check to a separate file and reuse kinda `new UrlParser()`

Reply:
I've just removed totally that requiring, so we don't need to introduce a package